source/extensions/resource_monitors/downstream_connections/downstream_connections_monitor.cc

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "extensions/resource_monitors/downstream_connections/downstream_connections_monitor.h"
+
+#include <thread>
+
+#include "envoy/config/resource_monitor/downstream_connections/v3/downstream_connections.pb.h"
+
+#include "common/common/assert.h"
+
+// tbd
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace DownstreamConnectionsMonitor {
+
+DownstreamConnectionsMonitor::DownstreamConnectionsMonitor(
+    const envoy::config::resource_monitor::fixed_heap::v3::DownstreamConnectionsConfig& config)
+    : max_downstream_conns_(config.max_downstream_connections()),
+      global_total_downstream_conns_(0) {
+  ASSERT(max_downstream_conns_ > 0);
+}
+
+void DownstreamConnectionsMonitor::updateResourceUsage(
+    Server::ResourceMonitor::Callbacks& callbacks) {
+  Server::ResourceUsage usage;
+  global_total_downstream_conns_.exchange(0);
```

Comment:
Is global_total_downstream_conns_ accessible from outside this method?

If no, it doesn't need to be a class member.  If yes, the exchange with 0 followed by the adds below may result in other threads getting an incorrect value for number of downstream connections if the atomic member is accessed in the middle of this computation.